core/psh/pshapp/pshapp.c

```diff
@@ -1143,14 +1143,10 @@ int psh_pshapp(int argc, char **argv)
 	}
 	/* Run shell interactively */
 	else {
-		if (pshapp_common.isrunpsh == 0) {
-			pshapp_common.isrunpsh = 1;
-			psh_run(1);
-			pshapp_common.isrunpsh = 0;
-		}
+		err = psh_run(1);
```

Comment:
We are already not complaint with one exit path rule in this function anyway
```suggestion
		return psh_run(1);
```

Reply:
Yes, right. Changed to `return`-ing directly.